pywbem/_logging.py

```diff
@@ -14,404 +14,341 @@
-    """
-    **Experimental:** *New in pywbem 0.11 as experimental.*
-
-    This metaclass allows the definition of __str__ and __repr__ at the
-    class level in the subclass.
+def configure_loggers_from_string(log_config_str, log_filename=None,
+                                  connection=None):
+    # pylint: disable=line-too-long
     """
-    loggers = {}
-
-    def __str__(cls):
-        return 'PywbemLoggers(loggers={s.loggers!r})'.format(s=cls)
+    Configure the pywbem loggers defined by the input string in the syntax
+    documented with the `log_config_str` parameter. This configures the defined
+    logger names with the destination and detail_level parameters defined by
+    the dest and detail_level components of the string.
 
-    def __repr__(cls):
-        return 'PywbemLoggers({s.loggers!r})'.format(s=cls)
+    Parameters:
 
+      log_config_str (:term:`string`): Specifies the logger definitions
+        with the following formats::
 
-# pylint: disable=no-init
-class PywbemLoggers(six.with_metaclass(MetaPywbemLoggers)):
-    """
-    **Experimental:** *New in pywbem 0.11 as experimental.*
-
-    Container for pywbem logger information when loggers are created. This
-    class is a singleton, there is only one set of data for a pywbem
-    instantiation.  Its goal is to defined named loggers from data input and
-    too record information about these named loggers in a dictionary for
-    use by the log functions
-
-    This is defined with only class level object and methods as an
-    easy way to create a singleton. However, at least in python 2.6, you
-    cannot make some magic methods work (ex.__getItem__)
-
-    There are two constructors for loggers:
-
-    * :meth:`~pywbem._logging.PywbemLoggers.create_loggers` - Creates one or
-      more logger entries in this class and also in the python logging class
-      from an input string with the format defined in the create_loggers
-      method below.
-
-    * :meth:`~pywbem._logging.PywbemLoggers.create_logger` - Creates a single
-      logger from the separate pywbem logging parameters supplied with the
-      method.
-    """
-
-    @classmethod
-    def create_loggers(cls, input_str, log_filename=None):
-        # pylint: disable=line-too-long
-        """
-        Create the pywbem loggers defined by the input string in the following
-        syntax and place the logger definitions in the class level dictionary
-        in this class.
-
-        Parameters:
-
-          input_str (:term:`string`): Specifies the logger definitions
-            as follows:
-
-            ``log_specs`` := ``log_spec`` [, ``log_spec`` ]
-
-            ``log_spec`` := ``log_comp`` ['=' [ ``dest`` ][":"[ ``detail_level`` ]]]]
-
+            log_specs := log_spec [, log_spec ]
+            log_spec := log_name ['=' [ dest ][":"[ detail_level ]]]]
             where:
+                log_name: Must be one of strings in the :data:`~pywbem._logging.LOG_SIMPLE_NAMES` list.
```

Comment:
For all elements, we should also state what the element means and not just which values are allowed.